corehq/apps/email/models.py

```diff
@@ -0,0 +1,153 @@
+        """
+        Returns all the backends that the given domain has access to.
+        """
+        domain_owned_backends = models.Q(domain=domain)
+        result = SQLEmailSMTPBackend.objects.filter(domain_owned_backends).distinct()
+
+        if count_only:
+            return result.count()
+
+        result = result.order_by('name').values_list('id', flat=True)
+
+        if offset is not None and limit is not None:
+            result = result[offset:offset + limit]
+
+        return [cls.load(pk) for pk in result]
+
+    @classmethod
+    def get_domain_default_backend(cls, domain, id_only=False):
+        result = SQLEmailSMTPBackend.objects.filter(
+            domain=domain,
+            is_default=True,
+        ).values_list(flat=True)
+
+        if len(result) > 1:
+            raise cls.MultipleObjectsReturned(
+                "More than one default backend found for "
+                "domain %s" % (domain)
+            )
+        elif len(result) == 1:
+            if id_only:
+                return result[0]
+            return cls.load(result[0])
+        return None
+
+    @staticmethod
+    def set_to_domain_default_backend(existing_default_backend, current_backend):
+        if existing_default_backend and current_backend:
+            existing_default_backend.is_default = False
+            existing_default_backend.save()
+
+            current_backend.is_default = True
+            current_backend.save()
+
+    @staticmethod
+    def unset_domain_default_backend(existing_default_backend):
+        if existing_default_backend:
+            existing_default_backend.is_default = False
+            existing_default_backend.save()
+
+    @classmethod
+    @quickcache(['backend_id'], timeout=60 * 60)
+    def get_backend_api_id(cls, backend_id):
+        filter_args = {'pk': backend_id}
+        result = SQLEmailSMTPBackend.objects.filter(**filter_args).values_list('hq_api_id', flat=True)
+
+        if len(result) == 0:
+            raise cls.DoesNotExist
+
+        return result[0]
+
+    @classmethod
+    @quickcache(['backend_id'], timeout=5 * 60)
+    def load(cls, backend_id, api_id=None):
+        """
+        backend_id - the pk of the SQLEmailSMTPBackend to load
+        api_id - if you know the hq_api_id of the SQLEmailSNTPBackend, pass it
+                 here for a faster lookup; otherwise, it will be looked up
+                 automatically
+        """
+        backend_classes = get_email_backend_classes()
+        api_id = api_id or cls.get_backend_api_id(backend_id)
+
+        if api_id not in backend_classes:
+            raise BadEmailConfigException("Unexpected backend api id found '%s' for "
+                                          "backend '%s'" % (api_id, backend_id))
+
+        klass = backend_classes[api_id]
+        result = klass.objects
+        return result.get(pk=backend_id)
+
+    @classmethod
+    def get_backend_from_id_and_api_id_result(cls, result):
```

Comment:
It looks like this method is not getting used.

Reply:
Yes, its no longer used. Removed.